pkg/endpoint/bpf.go

```diff
@@ -731,8 +761,18 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, err
 	// BPF programs for this endpoint.
 	if err = e.writeHeaderfile(epdir, owner); err != nil {
 		e.Mutex.Unlock()
-		return 0, fmt.Errorf("Unable to write header file: %s", err)
+		return 0, fmt.Errorf("unable to write header file: %s", err)
+	}
+
+	// Avoid BPF program compilation and installation if the headerfile hasn't changed.
+	bpfHeaderfileHash, err := e.hashHeaderfile(epdir)
+	if err != nil {
+		e.Mutex.Unlock()
+		return 0, fmt.Errorf("unable to hash header file: %s", err)
```

Comment:
I don't think we should fail just because the hash was not able to be calculated. Skipping the regeneration because the header is the same is just an optimization, so we should continue the regeneration but log the hash header failure.